packages/cadl-ranch-specs/http/type/model/visibility/main.tsp

```diff
@@ -30,6 +32,193 @@ model VisibilityModel {
   deleteProp: boolean;
 }
 
+@doc("Base model")
+model BaseModel {}
+
+@doc("Derived model")
+model DerivedModel extends BaseModel {
+  @doc("Required property")
+  resourceName: string;
+}
+
+@doc("Fixed string enum")
+enum FixedStringEnum {
+  One: "1",
+  Two: "2",
+  Four: "4",
+}
+
+@doc("Extensible enum")
+union ExtensibleEnum {
+  string,
+  One: "1",
+  Two: "2",
+  Four: "4",
+}
+
+@doc("RoundTrip model with readonly properties.")
+model RoundTripReadOnlyModel {
+  @doc("Required string, illustrating a readonly reference type property.")
```

Comment:
We can call it `ReadonlyModel` to indicate we only want to test the readonly properties.

All the properties inside don't have to have `readonly` as part of name because it already in the model name.